src/Mvc/Mvc.Core/src/ApplicationModels/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -47,7 +49,11 @@ internal class ApiBehaviorApplicationModelProvider : IApplicationModelProvider
 
         if (!options.SuppressInferBindingSourcesForParameters)
         {
-            ActionModelConventions.Add(new InferParameterBindingInfoConvention(modelMetadataProvider));
+            var serviceProviderIsService = serviceProvider.GetService<IServiceProviderIsService>();
+            var convention = options.DisableImplicitFromServicesParameters || serviceProviderIsService == null ?
```

Comment:
```suggestion
            var convention = options.DisableImplicitFromServicesParameters || serviceProviderIsService is null ?
```

Because https://github.com/dotnet/aspnetcore/issues/39323 is coming soon to a theatres near you